libs/feature/editor/src/lib/components/overview-upload/overview-upload.component.ts

```diff
@@ -2,12 +2,27 @@ import {
   ChangeDetectionStrategy,
   ChangeDetectorRef,
   Component,
+  EventEmitter,
   Input,
+  OnDestroy,
   OnInit,
+  Output,
 } from '@angular/core'
 import { CommonModule } from '@angular/common'
 import { RecordsApiService } from '@geonetwork-ui/data-access/gn4'
 import { UiInputsModule } from '@geonetwork-ui/ui/inputs'
+import { FormControl } from '@angular/forms'
+import { GraphicOverview } from '@geonetwork-ui/common/domain/model/record'
+import { Subject, takeUntil } from 'rxjs'
+
+const extractFileNameFormUrl = (url: URL, metadataUuid: string): string => {
+  const pattern = new RegExp(
+    `records/${metadataUuid}/attachments/([^/?#]+)(?:[/?#]|$)`,
+    'i'
+  )
+  const match = url.href.match(pattern)
+  return match ? match[1] : ''
+}
```

Comment:
I don't understand why this needed when you already have the resource fileName at your disposal...

Reply:
I don't know why this has been flagged as added code, maybe I moved it from another line. This not my code.

Reply:
it was definitely added by this PR, which makes sense since this file makes use of `extractFileNameFormUrl` and it wasn't available in that context. Maybe try factorizing the original function or simply exporting it?

Reply:
As seen with @jahow, this is needed finally.